src/main/java/com/lunark/lunark/service/AccountServiceImpl.java

```diff
@@ -39,8 +41,25 @@ public Account update(Account account) {
         return accountRepository.update(account);
     }
 
+    @Override public void delete(Long id) { accountRepository.delete(id); }
+
+
     @Override
-    public void delete(Long id) {
-        accountRepository.delete(id);
+    public Double getAverageGrade(Long id) {
+        Optional<Account> account = this.find(id);
+        if (account.isEmpty()) {
+            return -1.0;
```

Comment:
Would returning null here better convey that an account with that id does not exist?
```suggestion
            return null;
```